### DISCUSSION TITLE
How to hide header

### DISCUSSION THREAD

**Discussion by Malakay666** [breezy-weather/breezy-weather#1207] [Q&A]
Hello everyone, 
in changelog I found that header can be hidden somehow, but I can´t find it in the app itself. I have version 5.2.5 from F-Droid. Eventually could be size of temperature value set smaller somehow? 

Thanks for the help :)

**Comment by papjul** [comment]
Are you talking about this from changelog?

```
[Widgets] Text widget - Add ability to hide header (i.e. only show custom subtitle)
```

On the configuration screen of the Text widget, you have a "Hide header" switch. This will remove the temperature.

**Reply by Malakay666** [reply]
Ah, its widget related, I didn´t notice. I ment this header in app homescreen
![Screenshot_20240719-165156](https://github.com/user-attachments/assets/88decab5-5fae-43ed-bb09-47b2b91f6a87)
The best solution for me could be that temperature could be the same size as wind speed, humidity and uv index like this:
![Screenshot_20240719-165156](https://github.com/user-attachments/assets/3b89216e-a2e6-4bce-942e-cf2b9a8133fd)
Could this be done somehow?

**Reply by Malakay666** [reply]
My dislike is based on that the temperature itself consumes a lot of space on app homescreen and there is plenty of unused empty space around and everytime I have to scroll down. When there will be small temperature indicator instead, I will see more information I want.

**Reply by Malakay666** [reply]
I assume that this could be done by adding temperature indicator (like there is wind speed indicator etc) into the list and then it coul be managed like all other values.

**Reply by papjul** [reply]
There might be readability issues of putting white text info too high. For example, with white clouds with some weather backgrounds.

You also removed the "last refresh time" info.

It's also good to let the design breathe a bit. We had to find a compromise between users who wanted to "see more of the design" and users who wanted to jump directly in the data.

Regarding your mockup, it works with "Skoro jasno", but it's limited to ~10 characters before being cut, so it won't work most of the time. It also breaks the logic since the text is supposed to be "Temperature" like what is done with other indicators, and not the weather condition. I understand that this was just an idea though.

The more simple solution here would be to have a look at widgets which would be directly available on your homescreen. "Material You - Forecast" offer many information at one sight.

**Reply by Malakay666** [reply]
Yeah, but I prefer clean desktop with no widgets or icons on my phone. It keeps even battery last longer. So could be done some "tickable" setting to hide that temperature header completely?

**Reply by Malakay666** [reply]
And on your previous post - I only took screenshot and edited it - "Skoro jasno" its not necessary to have there, I ment only number.

**Reply by papjul** [reply]
I think battery consumption of widgets is a myth. Widgets are generated once. Every time the weather is updated in background, the widget is generated again with fresh data, and it doesn't change until next change of data. It's not actively consuming battery, it's like a static wallpaper.

Removing refresh time, current temperature and current weather condition is not an option. Showing them elsewhere can be considered.

**Reply by Malakay666** [reply]
All in all, could be added someday an option to completely hide header with some tickbox? I think that is easiest solution :)

**Reply by papjul** [reply]
Yes, that is indeed the easiest solution and the most specific to your need. However, as I said, we are not going to completely remove important data like refresh time, current temperature and current weather condition. Most weather apps have a similar homescreen layout as our app, and none offer the option to "hide header" as far as I know.

See also our policy for new preferences: https://github.com/breezy-weather/breezy-weather/blob/main/CONTRIBUTE.md#new-preference

**Reply by Malakay666** [reply]
Okay, can you tell me if it is possible to edit app homescreen somehow directly on my device? I assume that is something like a webpage, so i can hide it by editing code?

**Reply by papjul** [reply]
Is "directly on my device" a criteria? An Android app is compiled and packaged, it's not just one file you edit. You have to be able to run Gradle command lines to perform these actions and I don't think you can on an Android device. Even if you can, I think it would take a long time to build. I believe there are apps that can make smali patches to an existing app, but you would have to write them, and since this app is open source, it's just easier to just build it from source.
